src/transformers/modeling_tf_utils.py

```diff
@@ -2630,6 +2651,20 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
                         )
                         if resolved_archive_file is not None:
                             is_sharded = True
+                    if resolved_archive_file is None and filename == WEIGHTS_NAME:
+                        # Maybe the checkpoint is sharded, we try to grab the index name in this case.
+                        resolved_archive_file = cached_file(
+                            pretrained_model_name_or_path, WEIGHTS_INDEX_NAME, **cached_file_kwargs
+                        )
+                        if resolved_archive_file is not None:
+                            is_sharded = True
```

Comment:
Is this the entirety of supporting sharded PyTorch checkpoint -> TF models? Maybe make a PR with this first so we have it now and add Flax separately?

Reply:
Agreed, will revert on flax for simplicity!